src/pybind/mgr/prometheus/module.py

```diff
@@ -1012,14 +1012,21 @@ def get_df(self) -> None:
 
     @profile_method()
     def get_osd_blocklisted_entries(self) -> None:
-        _, out, _ = self.mon_command({
+        ret, out, err = self.mon_command({
             'prefix': 'osd blocklist ls',
             'format': 'json'
         })
-        combined_blocklists = json.loads(out)
-        blocklist = combined_blocklists[0]
-        range_blocklist = combined_blocklists[1]
-        blocklist_count = len(blocklist) + len(range_blocklist)
+       
+        try:
+          combined_blocklists = json.loads(out)
+          blocklist = combined_blocklists[0]
+          range_blocklist = combined_blocklists[1]
+          blocklist_count = len(blocklist) + len(range_blocklist)
+        except json.JSONDecodeError:
+          # Fallback for older Ceph monitor versions
```

Comment:
Yeah, that's the _alternative approach_ mentioned in the previous comment.

Actually, it's combined with the first one (conditional formatting). I'm not sure we need both.